### ISSUE TITLE
Extended Example: Bug or feature?

### ISSUE THREAD

**Issue by a-r-m-i-n** [b13/typo3-config#11]
I guess it is a bug in the documentation, here:
https://github.com/b13/typo3-config/blob/f5addfd528f8a4ba408e2ca8d7ba55a188cd28b9/examples/extended/AdditionalConfiguration.php#L29

In this example you extend \B13\Config by ProjectConfig to extend a new helper function (configureCustomLoggingForSomething).

You also use this helper function in the call below, but you initialized the `\B13\Config` class, instead of the extending `ProjectConfig` class.

Is there some magic wrapping/extending/proxy-loading mechanism existing, which automagically instantiates the extending class (`ProjectConfig`)? Or is it just a typo in the example? 😉 

Btw. thanks for publishing this library. Veeeery useful 👍

**Comment by bmack** [comment]
We use late-static-binding

https://github.com/b13/typo3-config/blob/main/src/Config.php#L69

So this means you need to call `initialize()` on the ProjectConfig and can then continue to work on it (that's how we do it actually)

**Comment by a-r-m-i-n** [comment]
Yes. But in this example you call initialize on the orignal class, not the extended one. Bug, right? ;)

**Comment by bmack** [comment]
Yes. You are correct.
